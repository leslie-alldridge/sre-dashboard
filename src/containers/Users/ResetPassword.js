import React, { Component } from "react";
import axios from "axios";

import TextField from "@material-ui/core/TextField";

import {
  LinkButtons,
  updateButton,
  homeButton,
  loginButton,
  HeaderBar,
  forgotButton,
  inputStyle,
  SubmitButtons
} from "../../components/common";

const loading = {
  margin: "1em",
  fontSize: "24px"
};

const title = {
  pageTitle: "Password Reset Screen"
};

export default class ResetPassword extends Component {
  constructor() {
    super();

    this.state = {
      username: "",
      password: "",
      confirmPassword: "",
      update: false,
      isLoading: true,
      error: false
    };
  }

  async componentDidMount() {
    await axios
      .get("/reset", {
        params: {
          resetPasswordToken: this.props.match.params.token
        }
      })
      .then(response => {
        if (response.data.message === "password reset link a-ok") {
          this.setState({
            username: response.data.username,
            update: false,
            isLoading: false,
            error: false
          });
        } else {
          this.setState({
            update: false,
            isLoading: false,
            error: true
          });
        }
      })
      .catch(error => {
        console.log(error.data);
      });
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  updatePassword = e => {
    e.preventDefault();
    axios
      .put("/updatepasswordviaemail", {
        username: this.state.username,
        password: this.state.password
      })
      .then(response => {
        if (response.data.message === "password updated") {
          this.setState({
            updated: true,
            error: false
          });
        } else {
          this.setState({
            updated: false,
            error: true
          });
        }
      })
      .catch(error => {
        console.log(error.data);
      });
  };

  render() {
    const { password, error, isLoading, updated } = this.state;

    if (error) {
      return (
        <div id="user-bg">
          <HeaderBar title={title} />
          <div style={loading}>
            <h4>Problem resetting password. Please send another reset link.</h4>
            <LinkButtons
              buttonText={`Go Home`}
              buttonStyle={homeButton}
              link={"/"}
            />
            <LinkButtons
              buttonStyle={forgotButton}
              buttonText={"Forgot Password?"}
              link={"/forgotPassword"}
            />
          </div>
        </div>
      );
    } else if (isLoading) {
      return (
        <div id="user-bg">
          <HeaderBar title={title} />
          <div style={loading}>Loading User Data...</div>
        </div>
      );
    } else {
      return (
        <div id="user-bg">
          <HeaderBar title={title} />
          <form className="password-form" onSubmit={this.updatePassword}>
            <TextField
              style={inputStyle}
              id="password"
              label="password"
              onChange={this.handleChange("password")}
              value={password}
              type="password"
            />
            <SubmitButtons
              buttonStyle={updateButton}
              buttonText={"Update Password"}
            />
          </form>

          {updated && (
            <div>
              <p>
                Your password has been successfully reset, please try logging in
                again.
              </p>
              <LinkButtons
                buttonStyle={loginButton}
                buttonText={"Login"}
                link={`/login`}
              />
            </div>
          )}
          <LinkButtons
            buttonText={`Go Home`}
            buttonStyle={homeButton}
            link={"/"}
          />
        </div>
      );
    }
  }
}
