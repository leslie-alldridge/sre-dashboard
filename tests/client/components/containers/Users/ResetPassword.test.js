import React from "react";
import { shallow } from "enzyme";
import ResetPassword from "../../../../../src/containers/Users/ResetPassword";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ResetPassword debug />);

    expect(component).toMatchSnapshot();
  });
});
