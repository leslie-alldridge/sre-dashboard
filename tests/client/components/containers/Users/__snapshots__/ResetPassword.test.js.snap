// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResetPassword
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "Password Reset Screen",
            }
          }
        />,
        <div
          style={
            Object {
              "fontSize": "24px",
              "margin": "1em",
            }
          }
        >
          Loading User Data...
        </div>,
      ],
      "id": "user-bg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "Password Reset Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading User Data...",
          "style": Object {
            "fontSize": "24px",
            "margin": "1em",
          },
        },
        "ref": null,
        "rendered": "Loading User Data...",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "Password Reset Screen",
              }
            }
          />,
          <div
            style={
              Object {
                "fontSize": "24px",
                "margin": "1em",
              }
            }
          >
            Loading User Data...
          </div>,
        ],
        "id": "user-bg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "Password Reset Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading User Data...",
            "style": Object {
              "fontSize": "24px",
              "margin": "1em",
            },
          },
          "ref": null,
          "rendered": "Loading User Data...",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
